app/Http/Controllers/GenreController.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace App\Http\Controllers;
+use App\Models\Genre;
+use Illuminate\Http\Request;
+
+class GenreController extends Controller
+{
+    private Genre $genre;
+
+    public function __construct(Genre $genre) 
+    {
+        $this->genre=$genre;
+    }
+
+    public function getAllGenres(Genre $genre) 
```

Comment:
No need to have `$genre` as a parameter here - you already have it in the constructor